main.go

```diff
@@ -155,13 +120,14 @@ func main() {
 		if dashboardURL != session.Request.URL.Path {
 			splittedUrl := strings.Split(session.Request.URL.Path, "/")
 			id := splittedUrl[len(splittedUrl)-2]
-			if val, ok := containers.Load(id); ok {
-				value := val.(SubscribedChan)
-				value.subCount = value.subCount - 1
-				containers.Store(id, val)
-				if value.subCount <= 0 {
-					infologger.Printf("Channel %s prepared to close\n", id)
-					containers.Delete(id)
+			if val, ok := containers[id]; ok {
+				val.Unsubscribe()
+				containers[id] = val
```

Comment:
I'm not sure how it works in go, but 10 lines before you added a new variable for same problem. If both pieces of code work, then it would be good to remove unnecessary variable.

Reply:
What do you mean? Which variable loads twice?